src/platform/android/ConfigurationManagerImpl.cpp

```diff
@@ -223,20 +223,6 @@ CHIP_ERROR ConfigurationManagerImpl::GetSoftwareVersionString(char * buf, size_t
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ConfigurationManagerImpl::GetHardwareVersionString(char * buf, size_t bufSize)
-{
-    CHIP_ERROR err;
-    size_t hardwareVersionLen = 0; // without counting null-terminator
-    err = ReadConfigValueStr(AndroidConfig::kConfigKey_HardwareVersionString, buf, bufSize, hardwareVersionLen);
```

Comment:
Hmm. Didn't know about that. Maybe we can add this exception to the legacy provider using some `#if CHIP_DEVICE_LAYER_TARGET_ANDROID`.